### PR TITLE
Allow user to select toolchain and display forc version in the compile window

### PR DESCRIPTION
**Pull request by sdankel**
- Adds the dropdown in the editor overlay to select toolchain version
- Displays the forc version in the compile output window
- beta-3 is the default toolchain version for now, so that it's compatible with what most people will use with fuel wallet.

to run this locally: 
1. uncomment [this line](https://github.com/FuelLabs/sway-playground/blob/a28d284ba28d46e99dacd0e9b2b6393fa838d211/app/src/features/editor/hooks/useCompile.tsx#L48) and comment out the line before it
2. from the root of the repo, run `cargo run`
3. in another terminal, `cd app && npm start`

https://github.com/FuelLabs/sway-playground/assets/47993817/ae7add29-29dd-4e72-a901-d422b5204997

### CODE REVIEW

File: app/src/features/editor/components/ToolchainDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import Tooltip from '@mui/material/Tooltip';
+import Select from '@mui/material/Select';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+
+const ToolchainNames = [
+  'latest',
+  'nightly',
```
Comment (JoshuaBatty) [inline app/src/features/editor/components/ToolchainDropdown.tsx:9]: Not sure if we should allow users to select the nightly option, how do you feel about just using `latest` along with the beta options?

File: src/compilation/mod.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            "".to_string(),
+        );
+    }
+
+    // Switch to the given fuel toolchain and check forc version.
+    switch_fuel_toolchain(toolchain);
+    let forc_version = check_forc_version();
+
+    // Create a new project.
+    let project_name = create_project().unwrap();
```
Comment (JoshuaBatty) [inline src/compilation/mod.rs:33]: should we change this to `expect` so if it panics, a message is printed to the console

File: app/src/features/editor/components/ToolchainDropdown.tsx (hunk 1 above)
Comment (sdankel) [reply app/src/features/editor/components/ToolchainDropdown.tsx:9]: We could, but why would we disallow nightly?

File: src/compilation/mod.rs (hunk 2 above)
Comment (sdankel) [reply src/compilation/mod.rs:33]: There is a lot of room for improvement regarding error handling in the server. Opened https://github.com/FuelLabs/sway-playground/issues/27

File: app/src/features/editor/components/ToolchainDropdown.tsx (hunk 1 above)
Comment (JoshuaBatty) [reply app/src/features/editor/components/ToolchainDropdown.tsx:9]: I was just thinking nightly might be a little too bleeding edge for the users but it's probably good to have it as an option. Can we have it show at the bottom of the list? Seems to be the rust playground approach. 

Screenshot 2023-08-15 at 9 24 15 am

Comment (sdankel) [reply app/src/features/editor/components/ToolchainDropdown.tsx]: Done, I put nightly and latest at the bottom.

**Review by JoshuaBatty — APPROVED** [review]
Nice, super handy feature!

**Pull request merged**